ts-client/src/amm/index.ts

```diff
@@ -0,0 +1,863 @@
+      tokenAAmount,
+      tokenBAmount,
+      poolLpSupply,
+      this.poolState.fees,
+    );
+
+    return {
+      poolTokenAmountOut: getMinAmountWithSlippage(poolTokenAmountOut, slippageRate),
+      tokenAInAmount,
+      tokenBInAmount,
+    };
+  }
+
+  public async deposit(owner: PublicKey, tokenAInAmount: BN, tokenBInAmount: BN, poolTokenAmount: BN) {
+    const { tokenAMint, tokenBMint, lpMint } = this.poolState;
+
+    const [[userAToken, createTokenAIx], [userBToken, createTokenBIx], [userPoolLp, createLpMintIx]] =
+      await this.createATAPreInstructions(owner, [tokenAMint, tokenBMint, lpMint]);
+
+    let preInstructions: Array<TransactionInstruction> = [];
+    createTokenAIx && preInstructions.push(createTokenAIx);
+    createTokenBIx && preInstructions.push(createTokenBIx);
+    createLpMintIx && preInstructions.push(createLpMintIx);
+
+    if (WRAPPED_SOL_MINT.equals(new PublicKey(this.tokenA.address))) {
+      preInstructions = preInstructions.concat(wrapSOLInstruction(owner, userAToken, tokenAInAmount.toNumber()));
+    }
+    if (WRAPPED_SOL_MINT.equals(new PublicKey(this.tokenB.address))) {
+      preInstructions = preInstructions.concat(wrapSOLInstruction(owner, userBToken, tokenBInAmount.toNumber()));
+    }
+
+    preInstructions.push(await this.getApySyncInstructions());
+
+    const postInstructions: Array<TransactionInstruction> = [];
+    if ([this.tokenA.address, this.tokenB.address].includes(WRAPPED_SOL_MINT.toBase58())) {
+      const closeWrappedSOLIx = await unwrapSOLInstruction(owner);
+      closeWrappedSOLIx && postInstructions.push(closeWrappedSOLIx);
+    }
+
+    const programMethod = this.isStablePool
+      ? this.program.methods.addImbalanceLiquidity
+      : this.program.methods.addBalanceLiquidity;
+
+    const depositTx = await programMethod(poolTokenAmount, tokenAInAmount, tokenBInAmount)
+      .accounts({
+        aTokenVault: this.vaultA.vaultState.tokenVault,
+        bTokenVault: this.vaultB.vaultState.tokenVault,
+        aVault: this.poolState.aVault,
+        bVault: this.poolState.bVault,
+        pool: this.address,
+        user: owner,
+        userAToken,
+        userBToken,
+        aVaultLp: this.poolState.aVaultLp,
+        bVaultLp: this.poolState.bVaultLp,
+        aVaultLpMint: this.vaultA.vaultState.lpMint,
+        bVaultLpMint: this.vaultB.vaultState.lpMint,
+        lpMint: this.poolState.lpMint,
+        tokenProgram: TOKEN_PROGRAM_ID,
+        vaultProgram: this.vaultProgram.programId,
+        userPoolLp,
+      })
+      .remainingAccounts(getRemainingAccounts(this.poolState))
+      .preInstructions(preInstructions)
+      .postInstructions(postInstructions)
+      .transaction();
+
+    return new Transaction({
+      feePayer: owner,
+      ...(await this.program.provider.connection.getLatestBlockhash('finalized')),
+    }).add(depositTx);
+  }
+
+  public async getWithdrawQuote(withdrawTokenAmount: BN, tokenMint?: PublicKey, slippage?: number) {
```

Comment:
Similar for this. I think we should add a comment for it. When tokenMint is null, withdrawTokenAmount will be the pool lp token. If it is not null, it must be either tokenA / token B of the pool

Reply:
Added jsdoc for all the public fn